contrib/pg_tde/documentation/docs/replication.md

```diff
@@ -1,4 +1,257 @@
+        -- Set this key as the active principal key
+        SELECT pg_tde_set_principal_key('my_vault_principal_key', 'vault');
+        ```
+
+2. **Standby:**
+    * In the standby's `postgresql.conf`, configure `pg_tde_key_provider` and all `pg_tde_vault_*` parameters, and `pg_tde_principal_key_name` to be **identical** to the primary's configuration.
+    * Ensure the standby server has network access to the Vault server and the necessary credentials/permissions to read the *same principal key* from Vault.
+    * The standby, upon startup, uses these settings to load the key.
+
+**Is it possible to use the same key provider and principal key?**
+
+**Yes, it is mandatory.** For physical replication to function with TDE, the standby *must* use the same principal key (by name and content, accessed via the same provider configuration) as the primary to decrypt the replicated data blocks.
+
+## Encrypted Table Replication
+
+**If a table on the primary site is encrypted, will all the standby sites also have the encrypted table replicated, given that the same key is implemented on the standby sites as well?**
+
+Yes.
+
+Physical replication works at the block level.
+
+1. When you encrypt a table on the primary using `CREATE TABLE ... USING heap_tde;` or by setting `default_table_access_method = 'heap_tde'`, the data blocks on disk are encrypted.
+2. Changes to this table (also encrypted) are written to WAL.
+3. Streaming replication sends these WAL records to the standby.
+4. The standby applies these WAL records, writing the encrypted blocks to its disk.
+5. If the standby has the same principal key available (via identical `pg_tde` configuration and key access), it can decrypt these blocks when you query the table on the standby.
+
+If the standby lacks the correct key, replicated data remains encrypted and unreadable, and queries will fail.
+
+## Setup Steps
+
+This assumes the `pg_tde` extension is available for installation on both primary and standby.
+
+### A. Primary Server Setup
+
+1. **Install the `pg_tde` extension files.**
+2. **Configure `postgresql.conf` for `pg_tde`:**
+
+    ```ini
+    shared_preload_libraries = 'pg_tde' // Add pg_tde, comma-separated if others exist
+
+    # Example for File Provider:
+    pg_tde_key_provider = 'file'
+    pg_tde_file_keys_path = '/var/lib/postgresql/pg_tde_keys'
+    pg_tde_principal_key_name = 'my_principal_key'
+
+    # Example for Vault Provider:
+    # pg_tde_key_provider = 'vault'
+    # pg_tde_vault_url = 'http://your-vault-server:8200'
+    # pg_tde_vault_token = 's.xxxxxxxxxxxxxx'
+    # pg_tde_vault_mount_path = 'secret/'
+    # pg_tde_principal_key_name = 'my_principal_key_in_vault'
+    ```
+
+3. **Configure `postgresql.conf` for Replication:**
+
+    ```ini
+    wal_level = replica
+    max_wal_senders = 5
+    archive_mode = on # Recommended
+    archive_command = 'cp %p /path/to/archive/%f' # Example
```

Comment:
shared_preload_libraries missing. And these settings have nothing to do with pg_tde, I don't think we should mention them.